momaland/learning/continuous_ppo/momappo_crazyrl.py

```diff
@@ -0,0 +1,541 @@
+from flax.training.train_state import TrainState
+from jax import vmap
+from supersuit import agent_indicator_v0, clip_actions_v0, normalize_obs_v0
+from tqdm import tqdm
+
+from momaland.envs.crazyrl.catch import catch_v0 as Catch
+from momaland.learning.continuous_ppo.utils import save_results
+from momaland.utils.env import ParallelEnv
+from momaland.utils.parallel_wrappers import (
+    LinearizeReward,
+    NormalizeReward,
+    RecordEpisodeStatistics,
+)
+
+
+def parse_args():
+    """Argument parsing for hyperparameter optimization."""
+    # fmt: off
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--exp-name", type=str, default=os.path.basename(__file__).rstrip(".py"),
+                        help="the name of this experiment")
+    parser.add_argument("--seed", type=int, default=1,
+                        help="seed of the experiment")
+    parser.add_argument("--debug", type=lambda x: bool(strtobool(x)), default=False, nargs="?", const=True, help="run in debug mode")
+
+    # Algorithm specific arguments
+    parser.add_argument("--num-steps", type=int, default=1280, help="the number of steps per epoch (higher batch size should be better)")
+    parser.add_argument("--total-timesteps", type=int, default=1e5,
+                        help="total timesteps of the experiments")
+    parser.add_argument("--update-epochs", type=int, default=2, help="the number epochs to update the policy")
+    parser.add_argument("--num-minibatches", type=int, default=2, help="the number of minibatches (keep small in MARL)")
+    parser.add_argument("--gamma", type=float, default=0.99,
+                        help="the discount factor gamma")
+    parser.add_argument("--lr", type=float, default=1e-3,
+                        help="the learning rate of the policy network optimizer")
+    parser.add_argument("--gae-lambda", type=float, default=0.99,
+                        help="the lambda for the generalized advantage estimation")
+    parser.add_argument("--clip-eps", type=float, default=0.2,
+                        help="the epsilon for clipping in the policy objective")
+    parser.add_argument("--ent-coef", type=float, default=0.0,
+                        help="the coefficient for the entropy bonus")
+    parser.add_argument("--vf-coef", type=float, default=0.8,
+                        help="the coefficient for the value function loss")
+    parser.add_argument("--max-grad-norm", type=float, default=0.5,
+                        help="the maximum norm for the gradient clipping")
+    parser.add_argument("--activation", type=str, default="tanh",
+                        help="the activation function for the neural networks")
+    parser.add_argument("--anneal-lr", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True,
+                        help="whether to anneal the learning rate linearly")
+
+    args = parser.parse_args()
+    # fmt: on
+    return args
+
+
+class Actor(nn.Module):
```

Comment:
We might want to make the actor and critic networks size and depth configurable if we want to apply it to multiple environments. Could we pass this as a list [256, 128, 128] for 3 layers of those sizes for example?